### ISSUE TITLE
[bug] Kubeflow pipeline doesn't pick-up metrics from file /mlpipeline-metrics.json

### ISSUE THREAD

**Issue by crbl1122** [kubeflow/pipelines#8424]
### Environment

*  How do you deploy Kubeflow Pipelines (KFP)?

*  KFP version:

*  KFP SDK version:

kfp                                   1.8.14
kfp-pipeline-spec              0.1.16
kfp-server-api                   1.8.5
### Steps to reproduce

I've ensured that both the metrics artifact is correctly named: "mlpipeline-metrics" and the file saved in the root of the container: "mlpipeline-metrics.json". Still the Kubeflow pipeline doesn't display the metrics.
I have seen this problem reported by others but seems to be a bug because no solution was documented. I am testing the metrics display in Kubeflow UI for a pipeline running on GCP. I use the reusable component method not python function based components. The component is containerized and pushed to artifact registry and I reference it through YAML specification. 

```
def produce_metrics(
    mlpipeline_metrics):

    accuracy = 0.9
    metrics = {
     'metrics': [{
      'name': 'accuracy-score', # 
      'numberValue':  accuracy, # 
      'format': "PERCENTAGE",   # 
    }]
  }
    
    # save to mlpipeline-metrics.json file in the root
    with open('/mlpipeline-metrics.json', 'w') as f:
        json.dump(metrics, f)

    # save to artifact path
    with open(mlpipeline_metrics + '.json', 'w') as f:
        json.dump(metrics, f)
    
def main_fn(arguments):
    
    training_table_bq = arguments.training_table
    validation_table_bq = arguments.validation_table
    schema_dict = generate_schema(training_table_bq)
    target_name = arguments.target_name
    gcs_model_path = arguments.gcs_model_path
    mlpipeline_metrics = arguments.mlpipeline_metrics
    
    # run train evaluate
    gcs_model_path = train_evaluate(training_table_bq, validation_table_bq, schema_dict, target_name, gcs_model_path, mlpipeline_metrics)
    
    return gcs_model_path
    
if __name__ == '__main__':
    
    parser = argparse.ArgumentParser(description = "train evaluate model")
    parser.add_argument("--training_table", type = str, help = 'Name of the input training table')
    parser.add_argument("--validation_table", type = str, help = 'Name of the input validation table')
    parser.add_argument("--target_name", type = str, help = 'Name of the target variable')
    parser.add_argument("--gcs_model_path", type = str, help = 'output directory where the model is saved')
    # parser.add_argument('--ui_metadata_output_path',
    #                   type=str,
    #                   default='/mlpipeline-ui-metadata.json',
    #                   help='Local output path for the file containing UI metadata JSON structure.')
    parser.add_argument('--mlpipeline_metrics',
                      type=str,
                      required = False,
                      default='/mlpipeline-metrics.json',
                      help='Local output path for the file containing metrics JSON structure.')
    
    
    args = parser.parse_args()
```

### Labels

---

Impacted by this bug? Give it a 👍.

**Comment by Linchin** [comment]
Hi @crbl1122, the correct name for the metrics file should be `mlpipeline-ui-metadata.json`. See reference here: https://www.kubeflow.org/docs/components/pipelines/v1/sdk/output-viewer/#v1-sdk-writing-out-metadata-for-the-output-viewers
Please let us know if you have any questions.

**Comment by crbl1122** [comment]
Hi @Linchin the above didn't work either. Please note that as written in the issue description, I am testing the metrics visualization from a YAML specified component, containerized in GCP Artifact registry (AR), not a function based component neither one defined through containerOp. So what I have is the containerized module existing in AR and the YAML specification file. The KBF version I printed above (v1) is from the virtual machine where I define the pipeline. I am not aware if once containerized and registered in AR, it is possible that a component to use a different kubeflow version?
Nevertheless, I was trying to follow the instructions from this document: [https://www.kubeflow.org/docs/components/pipelines/v1/sdk/pipelines-metrics/](url).
Now, I tried also the document you proposed with a tabular visualization of the sklearn's classification report and it didn't work either.

Here is the adjusted code:

`def table_vis(mlpipeline_ui_metadata_path, prediction, predict_proba, y_true, target_name):
    classif_report_dict = classification_report(y_true, prediction,\
                                         target_names=[target_name + '_0', target_name + '_1'], output_dict=True)
    print(classif_report_dict)
    report_df = pd.DataFrame(classif_report_dict)
    report_csv = '/mlpipeline-ui-metadata.json'
    # save
    report_df.to_csv(report_csv, index = False)
    
    print('print csv content')
    import sys
    with open(report_csv, 'r') as f:
        contents = f.read()
    print (contents)

    
    metadata = {
        'outputs' : [{
          'type': 'table',
          'storage': 'gcs',
          'format': 'csv',
          #'header': [x['name'] for x in schema],
          'header': ['precision', 'recall', 'f1-score', 'support'],
          'source': report_csv
        }]
      }
    
    with open(mlpipeline_ui_metadata_path, 'w') as metadata_file:
        json.dump(metadata, metadata_file)`

YAML specs:

`%%writefile training.yaml
name: training
description: Scikit trainer. Receives the name of train and validation BQ tables. Train, evaluate and save a model.

inputs:
- {name: training_table, type: String, description: 'name of the BQ training table'}
- {name: validation_table, type: String, description: 'name of the BQ validation table'}
- {name: target_name, type: String, description: 'Name of the target variable'}
- {name: max_depth, type: Integer, description: 'max depth'}
- {name: learning_rate, type: Float, description: 'learning rate'}
- {name: n_estimators, type: Integer, description: 'n estimators'}

outputs:
- {name:  gcs_model_path, type: OutputPath, description: 'output directory where the model is saved'}
# - {name:  MLPipeline_Metrics, type: Metrics, description: 'output directory where the metrics are saved'}
- {name:  MLPipeline UI Metadata, type: Metrics, description: 'output directory where the metrics are saved'}

implementation:
    container:
        image: ..... train_comp:latest
        command: [
            /src/component/train.py,
            --training_table, {inputValue: training_table},
            --validation_table, {inputValue: validation_table},
            --target_name, {inputValue: target_name},
            --gcs_model_path, {outputPath: gcs_model_path},
            # --mlpipeline_metrics_path, {outputPath: MLPipeline_Metrics},
            --mlpipeline_ui_metadata_path, {outputPath: MLPipeline UI Metadata},
            --max_depth, {inputValue: max_depth},
            --learning_rate, {inputValue: learning_rate},
            --n_estimators, {inputValue: n_estimators}
        ]`
